tests/AppBundle/EventListener/RedirectToPreferredLocaleListenerTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\AppBundle\EventListener;
+
+use AppBundle\EventListener\RedirectToPreferredLocaleListener;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Unit test for the event listener.
+ *
+ * See http://symfony.com/doc/current/book/testing.html#unit-tests
+ *
+ * Execute the application tests using this command (requires PHPUnit to be installed):
+ *
+ *     $ cd your-symfony-project/
+ *     $ ./vendor/bin/phpunit
+ *
+ * @author Michael COULLERET <michael.coulleret@gmail.com>
+ */
+class RedirectToPreferredLocaleListenerTest extends \PHPUnit_Framework_TestCase
+{
+    private $request;
+    private $urlGenerator;
+    private $event;
+    private $listener;
+
+    public function setUp()
+    {
+        $this->request = $this->prophesize(Request::class);
+        $this->urlGenerator = $this->prophesize(UrlGeneratorInterface::class);
+        $this->event = $this->prophesize(GetResponseEvent::class);
+        $this->listener = new RedirectToPreferredLocaleListener($this->urlGenerator->reveal(), 'en|fr|es');
+    }
+
+    public function testListenerOnKernelRequestUnexpectedValueException()
+    {
+        $this->setExpectedException(\UnexpectedValueException::class);
```

Comment:
I'm not agree, this alter the value in the ```$this->locales``` variable.
If I apply a correction I prefer to use ```array_filter```
- with empty locales : http://sandbox.onlinephpfunctions.com/code/00fd67fc9f3579190315a0e2d95f5a10e2d0f159
- with locales : http://sandbox.onlinephpfunctions.com/code/7ec7a8475191ce656e2b8d1b658a6b78960827a8